src/OrbitBase/include/OrbitBase/Sort.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_SORT_H_
+#define ORBIT_BASE_SORT_H_
+
+#include <algorithm>
+#include <functional>
+#include <type_traits>
+
+namespace orbit_base_internal {
+
+template <typename Comparator, typename Projection>
+static auto MakeComparator(Comparator comparator, Projection projection) {
+  return [comparator, projection](const auto& a, const auto& b) {
+    return comparator(projection(a), projection(b));
+  };
+}
+
+}  // namespace orbit_base_internal
+
+namespace orbit_base {
+
+template <typename RandomIt, typename Projection, typename Comparator = std::less<>,
+          typename = std::enable_if<
+              std::is_invocable_v<Projection, typename std::iterator_traits<RandomIt>::value_type>>>
+void sort(RandomIt first, RandomIt last, Projection projection = {}, Comparator comparator = {}) {
+  std::sort(first, last, orbit_base_internal::MakeComparator(comparator, projection));
+}
+
+template <typename RandomIt, typename Projection, typename Comparator = std::less<>,
+          typename = std::enable_if<
+              std::is_invocable_v<Projection, typename std::iterator_traits<RandomIt>::value_type>>>
+void stable_sort(RandomIt first, RandomIt last, Projection projection = {},
```

Comment:
Can we make these compatible with std::ranges::sort, meaning the comparator argument before the projection argument?

Reply:
I've thought about this. In theory, yes. But why would we? 
The arguments should be ordered in the order of decreasing probability of non-default value.
If the caller needs either of these overloads, uses the qualified id, adds an include, in 100% of cases, a projection will be passed. Non-default Comparator will be there only half if the cases.

Reply:
It would make it easier to move to std::ranges::sort in the future.

Reply:
(I don't feel strongly about this)

Reply:
> It would make it easier to move to std::ranges::sort in the future.

Yeah. But what is the worst problem we're gonna have when migrating? Fix it in ~20 places and if we forget smth, complier will happily point that out. Doesn't look too scary.